doc/src/manual/faq.md

```diff
@@ -732,6 +732,50 @@ julia> @sync for i in 1:3
 1 Foo  Bar 2 Foo  Bar 3 Foo  Bar
 ```
 
+## Arrays
+
+### What are the differences between zero-dimensional arrays and scalars?
+
+Zero-dimensional arrays are arrays of the form `Array{T,0}`. They behave similar
+to scalars, but there are important differences. They deserve a special mention
+because they are a special case which makes logical sense given the generic
+definition of arrays, but might be a bit unintuitive at first. The following
+line defines a zero-dimensional array:
+
+```
+julia> A = zeros()
+0-dimensional Array{Float64,0}:
+0.0
+```
+
+In this example, `A` is a mutable container that contains one element, which can
+be set by `A[] = 1.0` and retrieved with `A[]`. All zero-dimensional arrays have
+the same size (`size(A) == ()`), and length (`length(A) == 1`). In particular,
+zero-dimensional arrays are not empty. If you find this unintuitive, here are
+some ideas that might help to understand Julia's definition.
+
+* Zero-dimensional arrays are the "point" to vector's "line" and matrix's
+"plane". Just as a line has no area (but still represents a set of things), a
+point has no length or any dimensions at all (but still represents a thing).
+* We define `prod(())` to be 1, and the total number of elements in an array is
+the product of the size. The size of a zero-dimensional array is `()`, and
+therefore its length is `1`.
+* Zero-dimensional arrays don't natively have any dimensions into which you
+index -- they’re just `A[]`. We can apply the same "trailing one" rule for them
+as for all other array dimensionalities, so you can indeed index them as
+`A[1]`, `A[1,1]`, etc.
```

Comment:
I think  `A[1,1]` would produce an error

Reply:
It doesn't:
```jl
julia> A = fill(1)
0-dimensional Array{Int64,0}:
1

julia> A[1,1]
1
```
Same rule that allows indexing a vector like an `n×1` matrix (or `n×1×1` three-tensor):
```jl
julia> v = rand(3)
3-element Array{Float64,1}:
 0.7636187688715259
 0.6218914012928478
 0.22322554192055066

julia> v[3,1]
0.22322554192055066
```